ci-operator/step-registry/gcp/provision/bastionhost/gcp-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,227 @@
+        "contents": {
+          "source": "data:text/plain;base64,${SQUID_SH}"
+        },
+        "mode": 420
+      },
+      {
+        "path": "/etc/squid/proxy.sh",
+        "contents": {
+          "source": "data:text/plain;base64,${PROXY_SH}"
+        },
+        "mode": 420
+      }
+    ]
+  },
+  "systemd": {
+    "units": [
+      {
+        "contents": "[Unit]\nWants=network-online.target\nAfter=network-online.target\n[Service]\n\nStandardOutput=journal+console\nExecStart=bash /etc/squid.sh\n\n[Install]\nRequiredBy=multi-user.target\n",
+        "enabled": true,
+        "name": "squid.service"
+      },
+      {
+        "dropins": [
+          {
+            "contents": "[Service]\nExecStart=\nExecStart=/usr/lib/systemd/systemd-journal-gatewayd \\\n  --key=/opt/openshift/tls/journal-gatewayd.key \\\n  --cert=/opt/openshift/tls/journal-gatewayd.crt \\\n  --trust=/opt/openshift/tls/root-ca.crt\n",
+            "name": "certs.conf"
+          }
+        ],
+        "name": "systemd-journal-gatewayd.service"
+      }
+    ]
+  }
+}
+EOF
+}
+
+# TODO: move to image
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
+
+VPC_CONFIG="${SHARED_DIR}/customer_vpc_subnets.yaml"
+
+PROXY_IMAGE="registry.ci.openshift.org/origin/${OCP_RELEASE}:egress-http-proxy"
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+CLUSTER_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+REGION="${LEASED_RESOURCE}"
+echo "Using region: ${REGION}"
+echo "Cluster name: ${CLUSTER_NAME}"
+test -n "${REGION}"
+
+if [[ -z "${NETWORK}" || -z "${CONTROL_PLANE_SUBNET}" ]]; then
+  NETWORK=$(/tmp/yq r "${VPC_CONFIG}" 'platform.gcp.network')
+  CONTROL_PLANE_SUBNET=$(/tmp/yq r "${VPC_CONFIG}" 'platform.gcp.controlPlaneSubnet')
+fi
+if [[ -z "${NETWORK}" || -z "${CONTROL_PLANE_SUBNET}" ]]; then
+  echo "Could not find VPC network and control-plane subnet" && exit 1
+fi
+ZONE_0=$(gcloud compute regions describe ${REGION} --format=json | jq -r .zones[0] | cut -d "/" -f9)
+MACHINE_TYPE="n1-standard-1"
+
+curl -L -o /tmp/fcos-stable.json https://builds.coreos.fedoraproject.org/streams/stable.json
+IMAGE_NAME=$(jq -r .architectures.x86_64.images.gcp.name < /tmp/fcos-stable.json)
+if [ -z "${IMAGE_NAME}" ]; then
+  echo "Missing IMAGE in region: ${REGION}" 1>&2
+  exit 1
+fi
+IMAGE_PROJECT=$(jq -r .architectures.x86_64.images.gcp.project < /tmp/fcos-stable.json)
+IMAGE_RELEASE=$(jq -r .architectures.x86_64.images.gcp.release < /tmp/fcos-stable.json)
+echo "Using FCOS ${IMAGE_RELEASE} IMAGE: ${IMAGE_NAME}"
+
+ssh_pub_key=$(<"${CLUSTER_PROFILE_DIR}/ssh-publickey")
```

Comment:
This ssh key is the one from our gcp-qe's vault, we can login the bastion using our qe's key, right?

Reply:
Right.